src/graph.cpp

```diff
@@ -203,13 +213,94 @@ void Graph::augmentPath(const std::string &source, const std::string &target, co
+        choice = rand() % currentVertex->getAdj().size();
+        if(currentVertex->getAdj()[choice]->isSelected()){
+           currentVertex->getAdj()[choice]->setSelected(false);
+            currentVertex->getAdj()[choice]->getReverse()->setSelected(false);
+           deactivatedEdges.push_back(currentVertex->getAdj()[choice]);
+           }
+    }
+    return deactivatedEdges;
+}
+
+/**
+ * Takes a vector of edge pointers and sets the selected state of those edges and the corresponding reverses to false
+ * Time Complexity: O(size(edges))
+ * @param edges - Vector of edge pointers to be deactivated
+ * @return A vector of pointers for all edges that were deactivated
+ */
+std::vector<Edge*> Graph::deactivateEdges(std::vector<Edge *> edges) {
+    for(Edge* edge : edges){
+        edge->setSelected(false);
+        edge->getReverse()->setSelected(false);
+    }
+    return edges;
+}
 
 /**
-* Finds the stations that are at the end of the indicated station's line (i.e have only connection to one other station)
+ * Takes a vector of edge pointers and sets the selected state of those edges and the corresponding reverses to true
+ * Time Complexity: O(size(edges))
+ * @param edges - Vector of edge pointers to be activated
+ */
+void Graph::activateEdges(std::vector<Edge *> edges) {
+    for(Edge* edge : edges){
+        edge->setSelected(true);
+        edge->getReverse()->setSelected(true);
+    }
+}
+
+/**
+ * Calculates the maximum flow between a source vertex and a target vertex with (numEdges) number of random edges being deactivated and reactivated after calculating the maximum flow
+ * Time Complexity: O(|VE²|)
+ * @param numEdges - Number of edges to be deactivated and later reactivated
+ * @param source - Id of the source Vertex
+ * @param target - Id of the target Vertex
+ * @return The value of the Max Flow with the interrupted lines
+ */
+unsigned int Graph::maxFlowDeactivatedEdgesRandom(const int &numEdges, const std::string &source, const std::string &target) {
+    std::vector<Edge*> deactivatedEdges = deactivateEdges(numEdges);
+    unsigned int maxFlowInterrupted = edmondsKarp(source, target);
+    activateEdges(deactivatedEdges);
+    return maxFlowInterrupted;
+}
+
+/**
+ *Calculates the maximum flow between a source vertex and a target vertex with the edges inputted to the function being deactivated and reactivated after calculating the maximum flow
+ * Time Complexity: O(|VE²|)
+ * @param selectedEdges - Vector of edges to be deactivated and later reactivated
+ * @param source - Id of the source Vertex
+ * @param target - Id of the target Vertex
+ * @return The value of the Max Flow with the interrupted lines
+ */
+unsigned int Graph::maxFlowDeactivatedEdgesSelected(std::vector<Edge*> selectedEdges, const std::string &source, const std::string &target) {
+    std::vector<Edge*> deactivatedEdges = deactivateEdges(selectedEdges);
+    unsigned int maxFlowInterrupted = edmondsKarp(source, target);
```

Comment:
You should refactor this, as edmonds karp now takes a list of ids and not a single one, alter the parameters too